src/runtime/load_dynlib.cpp

```diff
@@ -41,4 +42,76 @@ extern "C" LEAN_EXPORT obj_res lean_load_dynlib(b_obj_arg path, obj_arg) {
         return io_result_mk_error(ex.what());
     }
 }
+
+/* loadPlugin : System.FilePath -> IO Unit */
+extern "C" LEAN_EXPORT obj_res lean_load_plugin(b_obj_arg path, obj_arg) {
+    // we never want to look up plugins using the system library search
+    std::string rpath;
+#if defined(LEAN_EMSCRIPTEN)
+    rpath = string_to_std(path);
+    auto sep = rpath.rfind('/');
+#elif defined(LEAN_WINDOWS)
+    constexpr unsigned BufferSize = 8192;
+    char buffer[BufferSize];
+    DWORD retval = GetFullPathName(string_cstr(path), BufferSize, buffer, nullptr);
+    if (retval == 0 || retval > BufferSize) {
+        rpath = string_to_std(path);
+    } else {
+        rpath = std::string(buffer);
+    }
+    auto sep = rpath.rfind('\\');
+#else
+    char buffer[PATH_MAX];
+    char * tmp = realpath(string_cstr(path), buffer);
+    if (tmp) {
+        rpath = std::string(tmp);
+    } else {
+        inc(path);
+        return io_result_mk_error(lean_mk_io_error_no_file_or_directory(path, ENOENT, mk_string("")));
+    }
+    auto sep = rpath.rfind('/');
+#endif
+    if (sep == std::string::npos) {
+        sep = 0;
+    } else {
+        sep++;
+    }
+    auto dot = rpath.rfind(".");
+    if (dot == std::string::npos) {
+        dot = rpath.size();
+    }
+    std::string pkg = rpath.substr(sep, dot - sep);
+    std::string sym = "initialize_" + pkg;
+    void * init;
+#ifdef LEAN_WINDOWS
+    HMODULE h = LoadLibrary(rpath.c_str());
+    if (!h) {
+        return io_result_mk_error((sstream()
+            << "error loading plugin " << rpath << ": " << GetLastError()).str());
+    }
+    init = reinterpret_cast<void *>(GetProcAddress(h, sym.c_str()));
+#else
+    // Like lean_load_dynlib, the library is loaded with RTLD_GLOBAL.
+    // This ensures the interpreter has access to plugin definitions that are also
+    // imported (e.g., an environment extension defined with builtin_initialize).
+    // It also means that loading multiple plugins which define the same symbol will error.
```

Comment:
```suggestion
    // In either case, loading the same symbol twice (and thus e.g. running initializers
    // manipulating global `IO.Ref`s twice) should be avoided; the common module
    // should instead be factored out into a separate shared library
```
I checked again and I don't think `dlopen` errors on duplicate symbols